client/src/main/java/org/spine3/base/MapStringifier.java

```diff
@@ -0,0 +1,201 @@
+     */
+    MapStringifier(Class<K> keyClass, Class<V> valueClass, String delimiter) {
+        super();
+        this.keyClass = keyClass;
+        this.valueClass = valueClass;
+        this.delimiter = ESCAPE_SEQUENCE + delimiter;
+    }
+
+    @Override
+    protected String toString(Map<K, V> obj) {
+        final StringBuilder stringBuilder = new StringBuilder(0);
+        for (Map.Entry<K, V> entry : obj.entrySet()) {
+            stringBuilder.append(entry.getKey())
+                         .append(KEY_VALUE_DELIMITER)
+                         .append(entry.getValue())
+                         .append(delimiter);
+        }
+        final int length = stringBuilder.length();
+        final String result = stringBuilder.substring(0, length - 2);
+        return result;
+    }
+
+    @Override
+    protected Map<K, V> fromString(String s) {
+        final String[] buckets = s.split(Pattern.quote(delimiter));
+        final Map<K, V> resultMap = newHashMap();
+
+        for (String bucket : buckets) {
+            final Map.Entry<K, V> convertedBucket = convert(bucket);
+            resultMap.put(convertedBucket.getKey(), convertedBucket.getValue());
+        }
+        return resultMap;
+    }
+
+    private Map.Entry<K, V> convert(String bucketToConvert) {
+        final String[] keyValue = bucketToConvert.split(Pattern.quote(KEY_VALUE_DELIMITER));
+        checkKeyValue(keyValue);
+
+        final String key = keyValue[0];
+        final String value = keyValue[1];
+
+        try {
+            final K convertedKey = convert(keyClass, key);
+            final V convertedValue = convert(valueClass, value);
+            final Map.Entry<K, V> convertedBucket =
+                    new AbstractMap.SimpleEntry<>(convertedKey, convertedValue);
+            return convertedBucket;
+        } catch (Throwable e) {
+            throw newIllegalArgumentException("The exception is occurred during the conversion", e);
+        }
+    }
+
+    @SuppressWarnings("unchecked") // It is OK because class is verified.
+    private static <I> I convert(Class<I> elementClass, String elementToConvert) {
+
+        if (isInteger(elementClass)) {
+            return (I) Ints.stringConverter()
+                           .convert(elementToConvert);
+        }
+
+        if (isLong(elementClass)) {
+            return (I) Longs.stringConverter()
+                            .convert(elementToConvert);
+        }
+
+        if (isString(elementClass)) {
+            return (I) elementToConvert;
+        }
+
+        final I convertedValue = Stringifiers.fromString(elementToConvert, elementClass);
+        return convertedValue;
+
+    }
+
+    private static void checkKeyValue(String[] keyValue) {
+        if (keyValue.length != 2) {
+            final String exMessage =
+                    "Illegal key-value format. The value should " +
+                    "be separated with a single `:` character.";
```

Comment:
Please have constant for the separator character and use it in the message. Consider avoiding back-ticks. You can name the character (colon) and then quote it.